rfcs/text/0030/process.yml

```diff
@@ -0,0 +1,191 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+- name: process
+  reusable:
+    top_level: true
+    expected:
+      - at: process
+        as: entry_leader
+        short_override: Information about the entry session leader process.
+      - at: process
+        as: session_leader
+        short_override: Information about the session leader process.
+      - at: process
+        as: group_leader
+        short_override: Information about the process group leader process. Undefined if same as process.
+
+  fields:
+
+    - name: ppid
+      level: core
+      type: long
+      description: >
+        Parent process id.
+      example: 4242
+
+    - name: sid
+      level: core
+      type: long
+      description: >
+        Session id.
+      example: 4242
+
+    - name: ppid_start
+      level: extended
+      type: date
+      example: "2016-05-23T08:05:34.853Z"
+      description: >
+        The time this process's parent started.
+
+    - name: pgid_start
+      level: extended
+      type: date
+      example: "2016-05-23T08:05:34.853Z"
+      description: >
+        The time this process's process group started.
+
+    - name: sid_start
+      level: extended
+      type: date
+      example: "2016-05-23T08:05:34.853Z"
+      description: >
+        The time this process's session started.
+
+    - name: previous_args
```

Comment:
[`process.parent.args`](https://www.elastic.co/guide/en/ecs/current/ecs-process.html#field-process-args) ?

Reply:
Note that parent.exe and parent.args are close to this but imprecise:
- the parent process could have exec'd a new program and its exe changed from when this child was forked)
- the child process may be exec'ing again in case previous_exe and previous_args are those of the child's last exec, not the parent process's when it forked.

This type of information is really valuable for spotting abuse of things by things like recent log4j2 issue.

Any more questions on this or can I resolve it?

Reply:
Makes sense to me. Resolve away :)